transfer.go

```diff
@@ -20,6 +20,34 @@ type TransferSourceType string
 // "account_frozen", "could_not_process", "bank_account_restricted", "invalid_currency".
 type TransferFailCode string
 
+// TransferDestinationType consts represent valid transfer destinations.
+type TransferDestinationType string
+
+const (
+	// TransferDestinationBankAccount is a constant representing a transfer destination
+	// which is a bank account.
+	TransferDestinationBankAccount TransferDestinationType = "bank_account"
+
+	// TransferDestinationCard is a constant representing a transfer destination
+	// which is a card.
+	TransferDestinationCard TransferDestinationType = "card"
+
+	// TransferDestinationAccount is a constant representing a transfer destination
+	// which is a Stripe account.
+	TransferDestinationAccount TransferDestinationType = "account"
+)
+
+// TransferDestination describes the destination of a Transfer.
+// The Type should indicate which object is fleshed out (eg. BankAccount or Card)
+// For more details see https://stripe.com/docs/api/go#transfer_object
+type TransferDestination struct {
+	Type        TransferDestinationType `json:"object"`
+	ID          string                  `json:"id"`
+	Card        *Card                   `json:"-"`
+	BankAccount *BankAccount            `json:"-"`
+	Account     *Account                `json:"-"`
```

Comment:
We need to get a real convention for ordering here, but for now can you just alphabetize this list?